.github/workflows/test_website.yml

```diff
@@ -0,0 +1,36 @@
+######################################
+## Custom Web Almanac GitHub action ##
+######################################
+#
+# This generates the chapters and tests the website when a pull request is
+# opened or added to on the original repo
+#
+name: Test Website
+on:
+  workflow_dispatch:
+  pull_request:
+    paths:
+      - src/**.js
+      - src/**.css
+      - src/**.html
+      - src/content/**.md
+      - src/**.py
+      - src/**.json
+      - src/requirements.txt
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    if: github.repository == 'HTTPArchive/almanac.httparchive.org'
+    steps:
+    - name: Checkout branch
+      uses: actions/checkout@v2
+    - name: Setup Node.js for use with actions
+      uses: actions/setup-node@v1
+      with:
+        version:  12.x
+    - name: Set up Python 3.7
+      uses: actions/setup-python@v2
+      with:
+        python-version: '3.7'
+    - name: Run the website
+      run: ./src/tools/scripts/run_and_test_website.sh
```

Comment:
Not sure why Docker will be any better or quicker? If we had a pre-built Docker image I'd tend to agree (but then also can't check any requirement changes that this action is also intended for!) but other than that seems a level of abstraction for no real gain here. Some people may prefer to develop that way, but GitHub Actions doesn't care so I don't see reason for the abstraction. Plus we do more than just set up the environment in this.

Reply:
The motivation for using Docker here would be to consolidate requirements and changes in one place rather than repeating it. However, we can revisit this later.